packages/@react-native-windows/cli/src/runWindows/utils/deploy.ts

```diff
@@ -275,45 +248,69 @@ export async function deployToDesktop(
     args.push('--direct-debugging', options.directDebugging.toString());
   }
 
-  await runPowerShellScriptFunction(
-    'Removing old version of the app',
-    windowsStoreAppUtils,
-    `Uninstall-App ${appName}`,
-    verbose,
-  );
-
   await runPowerShellScriptFunction(
     'Enabling Developer Mode',
     windowsStoreAppUtils,
     'EnableDevMode',
     verbose,
   );
 
+  const appPackageFolder = getAppPackage(options, projectName);
+
   if (options.release) {
+    await runPowerShellScriptFunction(
+        'Removing old version of the app',
+        windowsStoreAppUtils,
+        `Uninstall-App ${appName}`,
+        verbose,
+      ); 
+    const script = glob.sync(
+      path.join(appPackageFolder, 'Add-AppDevPackage.ps1'),
+    )[0];
     await runPowerShellScriptFunction(
       'Installing new version of the app',
       windowsStoreAppUtils,
       `Install-App "${script}" -Force`,
       verbose,
     );
   } else {
-    // Install the app package's dependencies before attempting to deploy.
-    await runPowerShellScriptFunction(
-      'Installing dependent framework packages',
-      windowsStoreAppUtils,
-      `Install-AppDependencies ${appxManifestPath} ${appPackageFolder} ${options.arch}`,
-      verbose,
-    );
-    await build.buildSolution(
-      buildTools,
-      slnFile,
-      options.release ? 'Release' : 'Debug',
-      options.arch,
-      {DeployLayout: 'true'},
-      verbose,
-      'deploy',
-      options.buildLogDirectory,
-    );
+    const appxRecipe = path.join(path.dirname(appxManifestPath), `${projectName}.build.appxrecipe`);
+    const IDE_folder = `${buildTools.installationPath}\\Common7\\IDE`;
+    const deployAppxRecipe_exe = `${IDE_folder}\\DeployAppRecipe.exe`;
+    if (fs.existsSync(deployAppxRecipe_exe)) {
+      // Need fix for VS's DeployAppxRecipe to return error code on failure and to be able to be run from any directory
+      const oldDir = process.cwd();
+      process.chdir(IDE_folder);
```

Comment:
We should fix `commandWithProgress` to forward a `cwd` to exec instead of changing the global process directory. Changing the global state breaks some assumptions, and if commandWithProgress throws, we've just mucked up some state something bad. (usually we want to wrap this in a `try/finally`.